### PR TITLE
Add keyup support to bindKeys

### PR DESCRIPTION
**Pull request by mgmarlow**
Allow `#bindKeys` to specify which handler to bind callbacks to. Accepted parameters are `keydown` or `keyup`.

```js
kontra.bindKeys(['up', 'down', 'left', 'right'], function(e) {
  console.log('key released: ', e)
}, 'keyup');
```

### CODE REVIEW

**Comment by straker** [comment]
Awesome, thanks for the PR. Question about the intent of how this should work. As the code is written, as long as someone never calls `keyReleased`, the key will never toggle to false. So you could press 10 keys, release them all, and then call `keyReleased` and all will return true.

Is this the intent or should the key only be considered released as long as another key is not pressed?

**Comment by mgmarlow** [comment]
No problem! That's a good callout--if `keyReleased` is called conditionally, I could see a problem arising with the scenario you posed.

I think the problem is that keys need to be tracked in the game loop to enable keeping track of the previously-pressed key. In this implementation `keyReleased` is clearing out that state, but like you said if it is called conditionally in a sprite's `update` function the previously-pressed key state will never be cleared.

I'm new to working with this library, but perhaps one consideration is exposing a function that works similarly to `this.draw` that is called within a sprite's `update` function.

**Note** that I've renamed the function to `keyPressedOnce` to clarify its purpose.

```js
import { updateKeys, keyReleased } from 'kontra'

const sprite = Sprite({
  // ...
  update (dt) {
    if (keyPressedOnce('left')) {
      // ...
    }

    updateKeyState()
  }
})
```

where we can change the definition as such to use previous state:

```js
export function keyPressedOnce(key) {
  return !!pressedKeys[key] && !previouslyPressedKeys[key]
}

export function updateKeyState() {
  previouslyPressedKeys = Object.assign({}, pressedKeys);
}
```

**Comment by straker** [comment]
No problem, thanks for being willing to contribute. If we need to keep track of the state during the game loop, you can use the `on(tick)` function and change the state there. The [pointer](https://github.com/straker/kontra/blob/main/src/pointer.js#L371-L379) already does that to keep track of object positions, we can do the same for the keyboard to keep track of released keys.

**Comment by mgmarlow** [comment]
Awesome, I'll take a look at `pointer` and update this PR.

**Comment by straker** [comment]
I'm not sure the function does what it says. For example, the game loop is written in a way that multiple `update` functions can be called in a single frame. This would mean that a function tied to `keyPressedOnce` would fire multiple times before the `tick_end` fired and reset it. 

I think the main problem is that keyUp is not a state you can look at like keypressed. A key is either pressed or not pressed, but you can't have a function that checks if the key was "just released." Instead keyup is an event you respond to, more like `bindKeys`. So maybe we don't have a function that you can use in update but rather a function like `bindKeys` but for keyup events? Maybe an option to `bindKeys` that says if the event fires on the down or the up of the key press.

**Comment by mgmarlow** [comment]
@straker Sounds good--I'll go that route. It does deviate a bit from the original PR, in that the keyup event is only observable outside of a sprite's `#update` function, but I think it's a step in the right direction.

File: src/keyboard.js [hunk 1]
```diff
@@ -129,8 +135,10 @@ export function initKeys() {
  *
  * @param {String|String[]} keys - Key or keys to bind.
  * @param {(evt: KeyboardEvent) => void} callback - The function to be called when the key is pressed.
+ * @param {'keydown'|'keyup'} handler - Whether to bind to keydown or keyup events. Default: 'keydown'.
```
Comment (straker) [inline src/keyboard.js:138]: The tests are failing in the tyepscript as this JSDoc is written as a required property. To make it optional use the `[]` syntax. If you also use the default sytanx of `=` you don't need to list the default in the comment:

```suggestion
 * @param {'keydown'|'keyup'} [handler=keydown] - Whether to bind to keydown or keyup events.
```

**Comment by straker** [comment]
Ya, sorry about that deviation. I took a look at [Phaser](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/keyboardevents/) and (from the notes I could find) they don't have a game loop function that checks the keyUp, so maybe it's a hard problem?

**Comment by straker** [comment]
Ops, looks like I just didn't know what to call it. They have a `justUp` function that looks similar to your first approach, so maybe the initial problem I brought up isn't a problem?

**Comment by mgmarlow** [comment]
Nice catch! I'll was looking through the Phaser docs earlier for just that.

I think outside of the original problem statement, it is still useful to have `#bindKeys` work with `keyup` events, mind if I break out a separate PR for addressing the `#justDown` equivalent?

**Comment by straker** [comment]
Yep, go for it. I think both are useful.

**Review by straker — APPROVED** [review]
Awesome work.

**Pull request merged**